### PR TITLE
Revert "Do something weird here just to demonstrate github"

### PR DESCRIPTION
**Pull request by MaureenHelm**
Reverts zephyriot/zephyr#1

### CODE REVIEW

File: kernel/microkernel/k_semaphore.c [hunk 1]
```diff
@@ -112,7 +112,34 @@ void _k_sem_group_wait_cancel(struct k_args *A)
...
 			} else {
 				S->waiters = X->next;
-				X->next /= 0;
+			}
+			if (X->Comm == _K_SVC_SEM_GROUP_WAIT_REQUEST
+			    || X->Comm == _K_SVC_SEM_GROUP_WAIT_READY) {
+				if (X->Comm == _K_SVC_SEM_GROUP_WAIT_READY) {
+					/* obtain struct k_args of waiting task */
+
+					struct k_args *waitTaskArgs = X->Ctxt.args;
```
Comment (galak) [inline kernel/microkernel/k_semaphore.c:121]: here

**Review by galak — CHANGES_REQUESTED** [review]

File: kernel/microkernel/k_semaphore.c (hunk 1 above)
Comment (galak) [reply kernel/microkernel/k_semaphore.c:121]: another comment

**Review by galak — APPROVED** [review]
Approved!

**Review by lpereira — CHANGES_REQUESTED** [review]
The divide by zero is fundamental, keep it.

File: kernel/microkernel/k_semaphore.c (hunk 1 above)
Comment (galak) [reply kernel/microkernel/k_semaphore.c:121]: once more

**Review by galak — CHANGES_REQUESTED** [review]
change

**Comment by nashif** [comment]
# Header
# Header 2

**Comment by nashif** [comment]
abandon

**Pull request closed without merging**